apps/taskman/service.py

```diff
@@ -107,10 +127,35 @@ def create_or_update_task(self, flaw: Flaw, fail_if_exists=False) -> Response:
         except JIRAError as e:
             return Response(data=e.response.json(), status=e.status_code)
 
-    def update_task_status(self, issue_key: str, status: TaskStatus) -> Response:
+    def update_task_status(
+        self,
+        issue_key: str,
+        status: TaskStatus,
+        resolution: Optional[TaskResolution] = None,
+    ) -> Response:
         """Transition a task to a new state"""
         try:
-            self.jira_conn.transition_issue(issue=issue_key, transition=status)
+            if resolution:
+                self.jira_conn.transition_issue(
+                    issue=issue_key,
+                    transition=status,
+                    resolution={"name": resolution},
+                )
+            else:
+                self.jira_conn.transition_issue(
+                    issue=issue_key,
+                    transition=status,
+                )
+
+            issue = self.jira_conn.issue(issue_key)
+            uuid_labels = [
+                label for label in issue.fields.labels if "flawuuid:" in label
+            ]
+            flaw_uuid = uuid_labels[0].split(":")[1]
+            flaw = Flaw.objects.get(uuid=flaw_uuid)
+            flaw.meta_attr["jira_task_status"] = status
+            flaw.meta_attr["jira_task_resolution"] = resolution
+            flaw.save()
```

Comment:
I am thinking about this metadata. It is definitely handy but this way it is not persistent. The issue is that on any flaw change it is synced from Bugzilla and the `meta_attr` is created from scratch. To make this working the Jira task metadata fetch would also have to happen on flaw sync in `collectors.bzimport.convertors.py`. You could either just remove this `meta_attr` creation for now or make it also a part of flaw sync. We should think about this a bit more - whether the Jira task information is needed in OSIDB and what eventually what parts are necessary. We could maybe start with no information in OSIDB (metadata does not matter) and just fetch everything on demand and we will see whether it is going to be reasonable in the future.

Reply:
I have have removed the validation from Flaw and added in the status change operation, please take a look again to evaluate if this would be a better solution of if we should discuss an alternative solution. Thank you! @osoukup